tests/nativeapp/test_manager.py

```diff
@@ -41,7 +55,7 @@ def test_sync_deploy_root_with_stage(
     mock_local_diff_with_stage, mock_stage_diff, mock_execute, temp_dir, mock_cursor
 ):
     mock_execute.return_value = mock_cursor([{"CURRENT_ROLE()": "old_role"}], [])
-    mock_diff_result = DiffResult()
+    mock_diff_result = DiffResult(different="setup.sql")
```

Comment:
To ensure that the diff logic is actually called due to the new `has_changes()` guard.